internal/approver/controller/test/suite.go

```diff
@@ -0,0 +1,173 @@
+			Kind:  "ClusterIssuer",
+			Group: "cert-manager.io",
+		}
+	)
+
+	JustBeforeEach(func() {
+		ctx, cancel = context.WithCancel(context.TODO())
+
+		var err error
+		cl, err = client.New(env.Config, client.Options{Scheme: api.Scheme})
+		Expect(err).NotTo(HaveOccurred())
+
+		namespace = corev1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				GenerateName: "test-csi-driver-spiffe-",
+			},
+		}
+		Expect(cl.Create(ctx, &namespace)).NotTo(HaveOccurred())
+
+		log := klogr.New().WithName("testing")
+		mgr, err := ctrl.NewManager(env.Config, ctrl.Options{
+			Scheme:                        api.Scheme,
+			LeaderElection:                true,
+			MetricsBindAddress:            "0",
+			LeaderElectionNamespace:       namespace.Name,
+			LeaderElectionID:              "cert-manager-csi-driver-spiffe-approver",
+			LeaderElectionReleaseOnCancel: true,
+			Logger:                        log,
+		})
+		Expect(err).NotTo(HaveOccurred())
+
+		Expect(controller.AddApprover(ctx, log, controller.Options{
+			Manager:   mgr,
+			Evaluator: evaluator,
+			IssuerRef: issuerRef,
+		})).NotTo(HaveOccurred())
+
+		By("Running Approver controller")
+		go mgr.Start(ctx)
+
+		By("Waiting for Leader Election")
+		<-mgr.Elected()
+
+		By("Waiting for Informers to Sync")
+		Expect(mgr.GetCache().WaitForCacheSync(ctx)).Should(BeTrue())
+	})
+
+	JustAfterEach(func() {
+		Expect(cl.Delete(ctx, &namespace)).NotTo(HaveOccurred())
+		cancel()
+	})
+
+	It("should ignore CertificateRequest that have the wrong IssuerRef", func() {
+		cr := cmapi.CertificateRequest{
+			ObjectMeta: metav1.ObjectMeta{
+				GenerateName: "cert-manager-csi-driver-spiffe-",
+				Namespace:    namespace.Name,
+			},
+			Spec: cmapi.CertificateRequestSpec{
+				IssuerRef: cmmeta.ObjectReference{
+					Name:  "not-spiffe-ca",
+					Kind:  "ClusterIssuer",
+					Group: "cert-manager.io",
+				},
+			},
+		}
+		Expect(cl.Create(ctx, &cr)).NotTo(HaveOccurred())
+
+		Consistently(func() bool {
+			Eventually(func() error {
+				return cl.Get(ctx, client.ObjectKeyFromObject(&cr), &cr)
+			}).Should(BeNil())
+			return apiutil.CertificateRequestIsApproved(&cr) || apiutil.CertificateRequestIsDenied(&cr)
+		}, "3s").Should(BeFalse(), "expected neither approved not denied")
+	})
+
+	It("should deny CertificateRequest when the evaluator returns error", func() {
+		evaluator.WithEvaluate(func(_ *cmapi.CertificateRequest) error {
+			return errors.New("this is an error")
+		})
+
+		cr := cmapi.CertificateRequest{
+			ObjectMeta: metav1.ObjectMeta{
+				GenerateName: "cert-manager-csi-driver-spiffe-",
+				Namespace:    namespace.Name,
+			},
+			Spec: cmapi.CertificateRequestSpec{
+				IssuerRef: issuerRef,
+			},
+		}
+		Expect(cl.Create(ctx, &cr)).NotTo(HaveOccurred())
+
+		Eventually(func() bool {
+			Eventually(func() error {
+				return cl.Get(ctx, client.ObjectKeyFromObject(&cr), &cr)
+			}).Should(BeNil())
+			return apiutil.CertificateRequestIsApproved(&cr)
+		}).Should(BeTrue(), "expected approval")
```

Comment:
Nit: (I see that the logic is fixed in following commit but not the description)
```suggestion
		}).Should(BeTrue(), "expected denial")
```